moveit_ros/planning_interface/move_group_interface/src/wrap_python_move_group.cpp

```diff
@@ -114,24 +114,25 @@ class MoveGroupInterfaceWrapper : protected py_bindings_tools::ROScppInitializer
   {
     moveit_msgs::RobotState msg;
     py_bindings_tools::deserializeMsg(state_str, msg);
-    robot_state::RobotState state(moveit::planning_interface::MoveGroupInterface::getJointValueTarget());
+    robot_state::RobotState state(moveit::planning_interface::MoveGroupInterface::getTargetRobotState());
     moveit::core::robotStateMsgToRobotState(msg, state);
     return moveit::planning_interface::MoveGroupInterface::setJointValueTarget(state);
   }
 
   bp::list getJointValueTargetPythonList()
   {
-    const robot_state::RobotState& values = moveit::planning_interface::MoveGroupInterface::getJointValueTarget();
+    std::vector<double> values;
+    MoveGroupInterface::getJointValueTarget(values);
     bp::list l;
-    for (const double *it = values.getVariablePositions(), *end = it + getVariableCount(); it != end; ++it)
-      l.append(*it);
+    for (const double value : values)
```

Comment:
```suggestion
    for (const double& value : values)
```

Reply:
It doesn't make sense to use references for plain types. This adds a superfluous dereference and doesn't save any copying effort.